platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -47,6 +47,46 @@ Note that certain features that rely on third-party tools, such as [PMD](https:/
 
 For details, see the full [compatibility documentation](userguide/compatibility.html#java).
 
+<a name="java-compiler-error-rendering"></a>
+### Java compiler errors in the failure report
+
+From now on, Gradle also includes Java compilation failures in the report at the end of the build.
+This will accompany the already existing errors reported when the task is executed.
+This improvement helps to identify compilation problems faster; with long executions, it can be hard to find the root cause of a compilation failure.
+
+The following is a sample output caused by a simple compilation failure:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation error (compilation:java:java-compilation-error)
+  sample-project/src/main/java/Problem.java:6: error: incompatible types: int cannot be converted to String
+          String a = 1;
+                     ^
+```
+
+If any warning happens during the compilation, it will also be included in the report:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation warning (compilation:java:java-compilation-warning)
+  sample-project/src/main/java/Problem1.java:6: warning: [cast] redundant cast to String
+          var warning = (String)"warning";
+                        ^
+Java compilation error (compilation:java:java-compilation-error)
+  sample-project/src/main/java/Problem2.java:6: error: incompatible types: int cannot be converted to String
+          String a = 1;
+                     ^
+```
+
+A limitation of this feature is that warnings are only reported when a failure occurs.
```

Comment:
It's not really a limitation, it's the scope of the feature. I propose we soften the sentence a little bit, by saying
```suggestion
Note, that the output remains unchanged when the Java compilation task finishes with warnings only. 
```
Also, this should be the last sentence of the release notes.

Reply:
I think the solution is somewhere in the middle: we should denote that if there is no failure, there are no warnings displayed.

Reply:
Pushed the new text, then resolve or comment